content/working_with/console/widgets/deploymentsView.md

```diff
@@ -0,0 +1,279 @@
+
+- buttons to execute a workflow on that deployment or invoke some other
+  deployment action
+
+What follows are different deployment-related widgets divided into 3 tabs:
+
+1. Last Execution - information about the last workflow execution for that
+   deployment
+2. Deployment Info - general information about the deployment
+3. History - a list of workflow executions for that deployment and a list of
+   logs
+
+## Environments and Services
+
+For clarity and better organization, deployments can form a hierarchical
+structure and be marked as:
+
+- environments - can hold other environments and services
+- services - the bottommost level of the hierarchy
+
+Marking the deployments happens by adding the following labels to the
+deployment:
+
+1. `csys-obj-parent` - the ID of the parent of the current deployment
+2. `csys-obj-type` - either `service` or `environment`
+   If there is no such label, the deployment is a service by default.
+
+Forming a hierarchical structure allows drilling down into deployments that are
+directly attached to a parent environment, as well as aggregating and
+propagating statuses of deployments for better problem discovery.
+
+### Deployment status
+
+A single deployment's status is:
+
+- _requires attention_ (red exclamation mark) if one of the following applies:
+  - at least one node is not successfully installed (and there is no workflow
+    in progress)
+  - the last workflow execution failed
+- _in progress_ (yellow spinner) if there is a workflow in progress
+- _good_ otherwise
+
+### Propagating deployment statuses
+
+If a deployment has child deployments, then their statuses are propagated so
+that:
+
+1. If there is at least nested deployment that requires attention, the parent's
+   status will be _requires attention_ too.
+2. Otherwise, if there is at least nested deployment that is in progress,
+   the parent's status will be _in progress_ too (unless the parent deployment
+   itself _requires attention_).
+
+### Drilling down to child deployments
+
+If a deployment has child deployments, the drill-down buttons in the top part of
+the right pane become active. Clicking them will navigate to a nested page that
+present the deployments (only environments or services, depending on the button
+clicked) from the next level in the hierarchy.
+
+The nested page has only a single Deployments View widget, but it can be
+customized using the
+[Edit Mode]({{< relref "working_with/console/customization/edit-mode.md" >}}).
+
+Drilling down can happen multiple times if there are more levels in the
+deployments hierarchy.
+
+To go back to the parent page, use the breadcrumbs in top part of the page.
+
+<!-- TODO: screenshot -->
+
+## Features
+
+### Filtering deployments
+
+Deployments visible in the table can be filtered in 3 ways:
+
+1. Typing in the search bar above the table
+2. Setting a default filter in the widget's configuration This filter will be
+   persisted and used when viewing the widget again.
```

Comment:
Probably a dot is missing here:

```suggestion
2. Setting a default filter in the widget's configuration. This filter will be
   persisted and used when viewing the widget again.
```

Reply:
Yup, again the autoformatting joined the lines :worried: Good catch